hazelcast/src/main/java/com/hazelcast/scheduledexecutor/IScheduledExecutorService.java

```diff
@@ -0,0 +1,395 @@
+                                                       long delay, TimeUnit unit);
+
+    /**
+     * Creates and executes a one-shot action that becomes enabled
+     * after the given delay on all {@link Member}s given.
+     *
+     * @param command the task to execute
+     * @param members the collections of members - where to execute the task
+     * @param delay the time from now to delay execution
+     * @param unit the time unit of the delay parameter
+     * @return a ScheduledFuture representing pending completion of
+     *         the task and whose {@code get()} method will return
+     *         {@code null} upon completion
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    <V> Map<Member, IScheduledFuture<V>> scheduleOnMembers(Callable<V> command, Collection<Member> members,
+                                                           long delay, TimeUnit unit);
+
+    /**
+     * Creates and executes a periodic action that becomes enabled first
+     * after the given initial delay, and subsequently with the given
+     * period on all {@link Member}s given. Executions will commence after
+     * {@code initialDelay} then {@code initialDelay+period}, then
+     * {@code initialDelay + 2 * period}, and so on.
+     * If any execution of this task
+     * takes longer than its period, then subsequent execution will be skipped.
+     *
+     * @param command the task to execute
+     * @param members the collections of members - where to execute the task
+     * @param initialDelay the time to delay first execution
+     * @param period the period between successive executions
+     * @param unit the time unit of the initialDelay and period parameters
+     * @return a ScheduledFuture representing pending completion of
+     *         the task, and whose {@code get()} method will throw an
+     *         exception upon cancellation
+     * @throws RejectedExecutionException if the task cannot be
+     *         scheduled for execution
+     * @throws NullPointerException if command is null
+     */
+    Map<Member, IScheduledFuture<?>> scheduleOnMembersWithRepetition(Runnable command, Collection<Member> members,
+                                                                     long initialDelay, long period, TimeUnit unit);
+
+    /**
+     * Creates a new {@link IScheduledFuture} from the given handler.
+     * This is useful in case your member node or client from which the original
+     * scheduling happened, went down, and now you want to access the <tt>ScheduledFuture</tt> again.
+     *
+     * @param handler The handler of the task as found from {@link IScheduledFuture#getHandler()}
+     * @param <V> The return type of callable tasks
+     * @return A new {@link IScheduledFuture} from the given handler.
+     */
+    <V> IScheduledFuture<V> getScheduled(ScheduledTaskHandler handler);
```

Comment:
Flavor: perhaps call this 'getScheduledFuture'? getScheduled doesn't say anything to me without looking at the documentation.